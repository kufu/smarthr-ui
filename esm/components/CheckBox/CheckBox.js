var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React, { useCallback } from 'react';
import styled, { css } from 'styled-components';
import { transparentize } from 'polished';
import { useTheme } from '../../hooks/useTheme';
import { Icon } from '../Icon';
export var CheckBox = function (_a) {
    var _b = _a.mixed, mixed = _b === void 0 ? false : _b, _c = _a.className, className = _c === void 0 ? '' : _c, onChange = _a.onChange, props = __rest(_a, ["mixed", "className", "onChange"]);
    var theme = useTheme();
    var checked = props.checked, disabled = props.disabled;
    var boxClassName = (checked ? 'active' : '') + " " + (disabled ? 'disabled' : '');
    var handleChange = useCallback(function (e) {
        if (onChange)
            onChange(e);
    }, [onChange]);
    return (React.createElement(Wrapper, { className: className, themes: theme },
        React.createElement(Input, __assign({}, props, { type: "checkbox", onChange: handleChange, themes: theme })),
        React.createElement(Box, { className: boxClassName, themes: theme }),
        checked && (React.createElement(IconWrap, { themes: theme },
            React.createElement(Icon, { name: mixed ? 'fa-minus' : 'fa-check', size: 10, color: "#fff" })))));
};
var Wrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  ", "\n"], ["\n  ",
    "\n"])), function (_a) {
    var themes = _a.themes;
    var size = themes.size;
    return css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n      position: relative;\n      display: inline-block;\n      width: ", ";\n      height: ", ";\n      flex-shrink: 0;\n      line-height: 1;\n      box-sizing: border-box;\n    "], ["\n      position: relative;\n      display: inline-block;\n      width: ", ";\n      height: ", ";\n      flex-shrink: 0;\n      line-height: 1;\n      box-sizing: border-box;\n    "])), size.pxToRem(16), size.pxToRem(16));
});
var Box = styled.span(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  ", "\n"], ["\n  ",
    "\n"])), function (_a) {
    var themes = _a.themes;
    var frame = themes.frame, palette = themes.palette;
    return css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n      position: absolute;\n      width: 100%;\n      height: 100%;\n      border-radius: 4px;\n      border: ", ";\n      background-color: #fff;\n      box-sizing: border-box;\n      pointer-events: none;\n      &.active {\n        border-color: ", ";\n        background-color: ", ";\n      }\n      &.disabled {\n        background-color: ", ";\n        border-color: ", ";\n        &.active {\n          border-color: ", ";\n        }\n      }\n    "], ["\n      position: absolute;\n      width: 100%;\n      height: 100%;\n      border-radius: 4px;\n      border: ", ";\n      background-color: #fff;\n      box-sizing: border-box;\n      pointer-events: none;\n      &.active {\n        border-color: ", ";\n        background-color: ", ";\n      }\n      &.disabled {\n        background-color: ", ";\n        border-color: ", ";\n        &.active {\n          border-color: ", ";\n        }\n      }\n    "])), frame.border.default, palette.MAIN, palette.MAIN, palette.BORDER, palette.BORDER, palette.BORDER);
});
var Input = styled.input(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n  ", "\n"], ["\n  ",
    "\n"])), function (_a) {
    var themes = _a.themes;
    var _b = themes.palette, MAIN = _b.MAIN, OUTLINE = _b.OUTLINE;
    return css(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n      opacity: 0;\n      position: absolute;\n      top: 0;\n      left: 0;\n      width: 100%;\n      height: 100%;\n      margin: 0;\n      cursor: pointer;\n      &[disabled] {\n        pointer-events: none;\n      }\n      &:hover + span {\n        box-shadow: 0 0 0 2px ", ";\n      }\n      &:focus + span {\n        box-shadow: 0 0 0 2px ", ";\n      }\n    "], ["\n      opacity: 0;\n      position: absolute;\n      top: 0;\n      left: 0;\n      width: 100%;\n      height: 100%;\n      margin: 0;\n      cursor: pointer;\n      &[disabled] {\n        pointer-events: none;\n      }\n      &:hover + span {\n        box-shadow: 0 0 0 2px ", ";\n      }\n      &:focus + span {\n        box-shadow: 0 0 0 2px ", ";\n      }\n    "])), transparentize(0.78, MAIN), OUTLINE);
});
var IconWrap = styled.span(templateObject_8 || (templateObject_8 = __makeTemplateObject(["\n  ", "\n"], ["\n  ",
    "\n"])), function (_a) {
    var themes = _a.themes;
    var size = themes.size;
    return css(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n      position: absolute;\n      top: 50%;\n      left: 50%;\n      display: inline-block;\n      width: ", ";\n      height: ", ";\n      transform: translate(-50%, -50%);\n      pointer-events: none;\n      & > svg {\n        vertical-align: top;\n      }\n    "], ["\n      position: absolute;\n      top: 50%;\n      left: 50%;\n      display: inline-block;\n      width: ", ";\n      height: ", ";\n      transform: translate(-50%, -50%);\n      pointer-events: none;\n      & > svg {\n        vertical-align: top;\n      }\n    "])), size.pxToRem(10), size.pxToRem(10));
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8;
//# sourceMappingURL=CheckBox.js.map