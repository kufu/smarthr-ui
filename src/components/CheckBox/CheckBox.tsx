import React, { VFC, useCallback } from 'react'
import styled, { css } from 'styled-components'
import { transparentize } from 'polished'

import { Theme, useTheme } from '../../hooks/useTheme'

import { FaCheckIcon, FaMinusIcon } from '../Icon'

export type Props = React.InputHTMLAttributes<HTMLInputElement> & {
  mixed?: boolean
}

export const CheckBox: VFC<Props> = ({ mixed = false, className = '', onChange, ...props }) => {
  const theme = useTheme()
  const { checked, disabled } = props
  const boxClassName = `${checked ? 'active' : ''} ${disabled ? 'disabled' : ''}`
  const handleChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      if (onChange) onChange(e)
    },
    [onChange],
  )

  return (
    <Wrapper className={className} themes={theme}>
      <Input
        {...props}
        type="checkbox"
        onChange={handleChange}
        themes={theme}
        {...(mixed && { 'aria-checked': 'mixed' })}
      />
      <Box className={boxClassName} themes={theme} />
      {checked && (
        <IconWrap themes={theme}>
          {mixed ? <FaMinusIcon size={10} color="#fff" /> : <FaCheckIcon size={10} color="#fff" />}
        </IconWrap>
      )}
    </Wrapper>
  )
}

const Wrapper = styled.span<{ themes: Theme }>`
  ${({ themes }) => {
    const { size } = themes
    return css`
      position: relative;
      display: inline-block;
      width: ${size.pxToRem(16)};
      height: ${size.pxToRem(16)};
      flex-shrink: 0;
      line-height: 1;
      box-sizing: border-box;
    `
  }}
`
const Box = styled.span<{ themes: Theme }>`
  ${({ themes }) => {
    const { frame, palette } = themes
    return css`
      position: absolute;
      width: 100%;
      height: 100%;
      border-radius: 4px;
      border: ${frame.border.default};
      background-color: #fff;
      box-sizing: border-box;
      pointer-events: none;
      &.active {
        border-color: ${palette.MAIN};
        background-color: ${palette.MAIN};
      }
      &.disabled {
        background-color: ${palette.BORDER};
        border-color: ${palette.BORDER};
        &.active {
          border-color: ${palette.BORDER};
        }
      }
    `
  }}
`
const Input = styled.input<{ themes: Theme }>`
  ${({ themes }) => {
    const { MAIN, OUTLINE } = themes.palette
    return css`
      opacity: 0;
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      margin: 0;
      cursor: pointer;
      &[disabled] {
        pointer-events: none;
      }
      &:hover + span {
        box-shadow: 0 0 0 2px ${transparentize(0.78, MAIN)};
      }
      &:focus + span {
        box-shadow: 0 0 0 2px ${OUTLINE};
      }
    `
  }}
`
const IconWrap = styled.span<{ themes: Theme }>`
  ${({ themes }) => {
    const { size } = themes

    return css`
      position: absolute;
      top: 50%;
      left: 50%;
      display: inline-block;
      width: ${size.pxToRem(10)};
      height: ${size.pxToRem(10)};
      transform: translate(-50%, -50%);
      pointer-events: none;
      & > svg {
        vertical-align: top;
      }
    `
  }}
`
