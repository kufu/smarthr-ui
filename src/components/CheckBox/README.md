# CheckBox

```tsx
import { CheckBox } from 'smarthr-ui'

<CheckBox name="sample" checked={true} mixed={true} onChange={() => {}} themeColor="light" />
```

## props

| Name       | Required | Type                                   | DefaultValue | Description                                                                                 |
| ---------- | -------- | -------------------------------------- | ------------ | ------------------------------------------------------------------------------------------- |
| checked    | ✓     | **boolean** <br> true &#124; false     | -            | Whether the check box is checked.                                                           |
| name       | ✓     | **string**                             | -            | the name attribute of input tag.                                                            |
| themeColor | -        | **string** <br>  'light' &#124; 'dark' | 'light'      | style theme.                                                                                |
| disabled   | -        | **boolean** <br> true &#124; false     | false        | The disabled attribute of input tag.                                                        |
| mixed      | -        | **boolean** <br> true &#124; false     | false        | If `true`, the check Icon change to minus.                                                  |
| className  | -        | **string**                             | ''           | The className attribute of component.                                                       |
| onChange   | -        | **function**                           | -            | Fired when the checkbox changed. <br><br>function: (name: string, checked: boolean) => void |
