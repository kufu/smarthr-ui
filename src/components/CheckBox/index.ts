export { CheckBox } from './CheckBox'
export type { Props } from './CheckBox'
