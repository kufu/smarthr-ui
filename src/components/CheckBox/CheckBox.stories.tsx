import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import styled from 'styled-components'

import { CheckBox } from './CheckBox'
import readme from './README.md'

const onChange = action('onChange')

storiesOf('CheckBox', module)
  .addParameters({
    readme: {
      sidebar: readme,
    },
  })
  .add('all', () => (
    <Group>
      <li>
        <Text>checked</Text>
        <List>
          <li>
            <CheckBox name="sample" checked={true} onChange={onChange} />
          </li>
          <li>
            <CheckBox name="sample" checked={true} disabled={true} onChange={onChange} />
          </li>
        </List>
      </li>

      <li>
        <Text>unchecked</Text>
        <List>
          <li>
            <CheckBox name="sample" checked={false} onChange={onChange} />
          </li>
          <li>
            <CheckBox name="sample" checked={false} disabled={true} onChange={onChange} />
          </li>
        </List>
      </li>

      <li>
        <Text>mixed</Text>
        <List>
          <li>
            <CheckBox name="sample" checked={true} mixed={true} onChange={onChange} />
          </li>
          <li>
            <CheckBox
              name="sample"
              checked={true}
              mixed={true}
              disabled={true}
              onChange={onChange}
            />
          </li>
        </List>
      </li>
    </Group>
  ))

const List = styled.ul`
  padding: 0;
  & > li {
    display: inline-block;
    padding: 16px;
    &:not(:first-child) {
      margin-left: 16px;
    }
    &.dark {
      background-color: gray;
    }
  }
`
const Group = styled.ul`
  list-style: none;
  & > li:not(:first-child) {
    margin-top: 24px;
  }
`
const Text = styled.p`
  margin: 0 0 16px 0;
  font-size: 16px;
`
