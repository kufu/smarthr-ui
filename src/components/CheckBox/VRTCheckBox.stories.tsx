import { StoryFn } from '@storybook/react'
import React, { ChangeEvent, useState } from 'react'
import styled from 'styled-components'

import { InformationPanel } from '../InformationPanel'

import { All } from './CheckBox.stories'

import { CheckBox } from '.'

export default {
  title: 'Forms（フォーム）/CheckBox',
  component: CheckBox,
  parameters: {
    withTheming: true,
  },
}

export const VRTState: StoryFn = () => {
  const [checkedName, setCheckedName] = useState<string[]>([])
  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const currentName = e.currentTarget.name

    if (checkedName.includes(currentName)) {
      setCheckedName(checkedName.filter((name) => name !== currentName))
    } else {
      setCheckedName([...checkedName, currentName])
    }
  }

  return (
    <WrapperList>
      <VRTInformationPanel title="VRT 用の Story です" togglable={false}>
        hover, activeなどの状態で表示されます
      </VRTInformationPanel>
      <li>
        <Title>hover</Title>
        <InnerList id="list-hover">
          <li>
            <CheckBox name="2" checked={checkedName.includes('2')} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" checked={true} error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} mixed onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="disabled" disabled onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="checked_disabled" checked disabled onChange={handleChange} />
          </li>
        </InnerList>
      </li>

      <li>
        <Title>focus</Title>
        <InnerList id="list-focus">
          <li>
            <CheckBox name="2" checked={checkedName.includes('2')} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" checked={true} error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} mixed onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="disabled" disabled onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="checked_disabled" checked disabled onChange={handleChange} />
          </li>
        </InnerList>
      </li>

      <li>
        <Title>focus-visible</Title>
        <InnerList id="list-focus-visible">
          <li>
            <CheckBox name="2" checked={checkedName.includes('2')} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" checked={true} error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} mixed onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="disabled" disabled onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="checked_disabled" checked disabled onChange={handleChange} />
          </li>
        </InnerList>
      </li>

      <li>
        <Title>active</Title>
        <InnerList id="list-active">
          <li>
            <CheckBox name="2" checked={checkedName.includes('2')} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="error" checked={true} error onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="2" checked={true} mixed onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="disabled" disabled onChange={handleChange} />
          </li>
          <li>
            <CheckBox name="checked_disabled" checked disabled onChange={handleChange} />
          </li>
        </InnerList>
      </li>
    </WrapperList>
  )
}

export const VRTForcedColors: StoryFn = () => (
  <>
    <VRTInformationPanel title="VRT 用の Story です" togglable={false}>
      Chromatic 上では強制カラーモードで表示されます
    </VRTInformationPanel>
    <All />
  </>
)

VRTState.parameters = {
  controls: { hideNoControlsWarning: true },
  pseudo: {
    hover: ['#list-hover input'],
    focus: ['#list-focus input'],
    focusVisible: ['#list-focus-visible input'],
    active: ['#list-active input'],
  },
}
VRTForcedColors.parameters = {
  chromatic: { forcedColors: 'active' },
}

const WrapperList = styled.ul`
  padding: 0 24px;
  list-style: none;
  & > li {
    padding: 16px;
    &:not(:first-child) {
      margin-top: 8px;
    }
  }
`
const InnerList = styled.ul`
  padding: 0;
  list-style: none;
  & > li {
    display: inline-block;
    &:not(:first-child) {
      margin-left: 16px;
    }
  }
`
const Title = styled.p`
  margin: 0 0 16px;
`

const VRTInformationPanel = styled(InformationPanel)`
  margin-bottom: 24px;
`
