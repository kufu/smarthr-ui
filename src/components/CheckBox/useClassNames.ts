import { useMemo } from 'react'

import { useClassNameGenerator } from '../../hooks/useClassNameGenerator'
import { CheckBox } from './CheckBox'

export function useClassNames() {
  const generate = useClassNameGenerator(CheckBox.displayName || 'CheckBox')
  return useMemo(
    () => ({
      wrapper: generate(),
    }),
    [generate],
  )
}
