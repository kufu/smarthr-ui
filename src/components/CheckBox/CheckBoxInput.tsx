import React, { VFC, useCallback } from 'react'
import styled, { css } from 'styled-components'
import { transparentize } from 'polished'

import { Theme, useTheme } from '../../hooks/useTheme'
import { useClassNames } from './useClassNames'

import { FaCheckIcon, FaMinusIcon } from '../Icon'

export type Props = React.InputHTMLAttributes<HTMLInputElement> & {
  /** `true` のとき、チェック状態を `mixed` にする */
  mixed?: boolean
}

export const CheckBoxInput: VFC<Props> = ({ mixed = false, onChange, ...props }) => {
  const theme = useTheme()
  const { checked, disabled } = props
  const boxClassName = `${checked ? 'active' : ''} ${disabled ? 'disabled' : ''}`
  const handleChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      if (onChange) onChange(e)
    },
    [onChange],
  )
  const classNames = useClassNames()

  return (
    <Wrapper themes={theme}>
      <Input
        {...props}
        type="checkbox"
        onChange={handleChange}
        className={classNames.checkBox}
        themes={theme}
        {...(mixed && { 'aria-checked': 'mixed' })}
      />
      <Box className={boxClassName} themes={theme} />
      {checked && (
        <IconWrap themes={theme}>
          {mixed ? <FaMinusIcon color="TEXT_WHITE" /> : <FaCheckIcon color="TEXT_WHITE" />}
        </IconWrap>
      )}
    </Wrapper>
  )
}

const Wrapper = styled.span<{ themes: Theme }>`
  ${({ themes }) => {
    const { fontSize } = themes
    return css`
      position: relative;
      display: inline-block;
      width: ${fontSize.M};
      height: ${fontSize.M};
      flex-shrink: 0;
      line-height: 1;
      box-sizing: border-box;
    `
  }}
`
const Box = styled.span<{ themes: Theme }>`
  ${({ themes }) => {
    const { border, color } = themes
    return css`
      position: absolute;
      width: 100%;
      height: 100%;
      border-radius: 4px;
      border: ${border.shorthand};
      background-color: ${color.WHITE};
      box-sizing: border-box;
      pointer-events: none;
      &.active {
        border-color: ${color.MAIN};
        background-color: ${color.MAIN};
      }
      &.disabled {
        background-color: ${color.BORDER};
        border-color: ${color.BORDER};
        &.active {
          border-color: ${color.BORDER};
        }
      }
    `
  }}
`
const Input = styled.input<{ themes: Theme }>`
  ${({ themes }) => {
    const { shadow, color } = themes
    return css`
      opacity: 0;
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      margin: 0;
      cursor: pointer;
      &[disabled] {
        pointer-events: none;
      }
      &:hover + span {
        box-shadow: 0 0 0 2px ${transparentize(0.78, color.MAIN)};
      }
      &:focus + span {
        box-shadow: ${shadow.OUTLINE};
      }
    `
  }}
`
const IconWrap = styled.span<{ themes: Theme }>`
  ${({ themes }) => {
    const { fontSize } = themes

    return css`
      position: absolute;
      top: 50%;
      left: 50%;
      display: inline-block;
      width: ${fontSize.XXS};
      height: ${fontSize.XXS};
      font-size: ${fontSize.XXS};
      transform: translate(-50%, -50%);
      pointer-events: none;
      & > svg {
        vertical-align: top;
      }
    `
  }}
`
