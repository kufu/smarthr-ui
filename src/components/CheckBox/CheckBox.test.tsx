import { create } from 'react-test-renderer'
import React from 'react'
import ReactDOM from 'react-dom'

import { CheckBox } from './CheckBox'

describe('CheckBox', () => {
  let container: HTMLDivElement
  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })
  afterEach(() => {
    document.body.removeChild(container)
  })
  it('should be match snapshot', () => {
    const testRenderer = create(<CheckBox name="sample" checked={true} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  it('should be render with value of name and checked attribute', () => {
    const name = 'sample'
    const checked = false
    ReactDOM.render(<CheckBox name={name} checked={checked} />, container)

    expect(document.querySelector('input')!.name).toBe(name)
    expect(document.querySelector('input')!.checked).toBe(checked)
  })

  it('should be render check icon if checked attribute is true', () => {
    const name = 'sample'
    const checked = true
    ReactDOM.render(<CheckBox name={name} checked={checked} />, container)

    expect(document.querySelectorAll('svg')).toHaveLength(1)
    expect(document.querySelector('input')!.checked).toBe(checked)
  })

  it('should be not render check icon if checked attribute is false', () => {
    const name = 'sample'
    const checked = false
    ReactDOM.render(<CheckBox name={name} checked={checked} />, container)

    expect(document.querySelectorAll('svg')).toHaveLength(0)
    expect(document.querySelector('input')!.checked).toBe(checked)
  })

  it('should be call onChange with name and checked', () => {
    const name = 'sample'
    const checked = true
    let target: HTMLInputElement
    const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      target = e.target
    }
    ReactDOM.render(<CheckBox name={name} checked={checked} onChange={onChange} />, container)
    container.querySelector('input')!.click()
    // @ts-expect-error
    expect(target.name).toBe('sample')
    // @ts-expect-error
    expect(target.checked).toBeTruthy()
  })
})
