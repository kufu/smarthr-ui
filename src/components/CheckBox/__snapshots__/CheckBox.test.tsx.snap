// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should be match snapshot 1`] = `
<Component
  checked={true}
  name="sample"
>
  <styled.div
    className=""
    themes={
      Object {
        "frame": Object {
          "border": Object {
            "default": "1px solid #d6d6d6",
            "lineStyle": "solid",
            "lineWidth": "1px",
            "radius": Object {
              "m": "6px",
              "s": "4px",
            },
          },
        },
        "interaction": Object {
          "hover": Object {
            "animation": ".3s ease-out",
            "animationDuration": ".3s",
            "animationTiming": "ease-out",
            "feedbackOpacity": ".7",
          },
        },
        "palette": Object {
          "BACKGROUND": "#f5f6fa",
          "BORDER": "#d6d6d6",
          "BRAND": "#00c4cc",
          "COLUMN": "#f9f9f9",
          "DANGER": "#ef475b",
          "HEADER_GREEN": "#57d0d5",
          "MAIN": "#00a5ab",
          "OUTLINE": "rgba(0,165,171,0.5)",
          "OVERLAY": "rgba(0,0,0,0.15)",
          "SCRIM": "rgba(0,0,0,0.5)",
          "TEXT_BLACK": "#333",
          "TEXT_DISABLED": "#c1c1c1",
          "TEXT_GREY": "#767676",
          "TEXT_LINK": "#007bc2",
          "WARNING": "#ff8800",
          "disableColor": [Function],
          "hoverColor": [Function],
        },
        "size": Object {
          "font": Object {
            "GRANDE": 18,
            "SHORT": 11,
            "TALL": 14,
            "VENTI": 24,
          },
          "mediaQuery": Object {
            "SP": 599,
            "TABLET": 959,
          },
          "pxToRem": [Function],
          "space": Object {
            "L": 40,
            "M": 32,
            "S": 24,
            "XL": 48,
            "XS": 16,
            "XXL": 56,
            "XXS": 8,
          },
        },
      }
    }
  >
    <div
      className="sc-AxirZ clMtsN"
    >
      <styled.input
        checked={true}
        name="sample"
        onChange={[Function]}
        themes={
          Object {
            "frame": Object {
              "border": Object {
                "default": "1px solid #d6d6d6",
                "lineStyle": "solid",
                "lineWidth": "1px",
                "radius": Object {
                  "m": "6px",
                  "s": "4px",
                },
              },
            },
            "interaction": Object {
              "hover": Object {
                "animation": ".3s ease-out",
                "animationDuration": ".3s",
                "animationTiming": "ease-out",
                "feedbackOpacity": ".7",
              },
            },
            "palette": Object {
              "BACKGROUND": "#f5f6fa",
              "BORDER": "#d6d6d6",
              "BRAND": "#00c4cc",
              "COLUMN": "#f9f9f9",
              "DANGER": "#ef475b",
              "HEADER_GREEN": "#57d0d5",
              "MAIN": "#00a5ab",
              "OUTLINE": "rgba(0,165,171,0.5)",
              "OVERLAY": "rgba(0,0,0,0.15)",
              "SCRIM": "rgba(0,0,0,0.5)",
              "TEXT_BLACK": "#333",
              "TEXT_DISABLED": "#c1c1c1",
              "TEXT_GREY": "#767676",
              "TEXT_LINK": "#007bc2",
              "WARNING": "#ff8800",
              "disableColor": [Function],
              "hoverColor": [Function],
            },
            "size": Object {
              "font": Object {
                "GRANDE": 18,
                "SHORT": 11,
                "TALL": 14,
                "VENTI": 24,
              },
              "mediaQuery": Object {
                "SP": 599,
                "TABLET": 959,
              },
              "pxToRem": [Function],
              "space": Object {
                "L": 40,
                "M": 32,
                "S": 24,
                "XL": 48,
                "XS": 16,
                "XXL": 56,
                "XXS": 8,
              },
            },
          }
        }
        type="checkbox"
      >
        <input
          checked={true}
          className="sc-AxhCb bzdyKr"
          name="sample"
          onChange={[Function]}
          type="checkbox"
        />
      </styled.input>
      <styled.span
        className="active "
        themes={
          Object {
            "frame": Object {
              "border": Object {
                "default": "1px solid #d6d6d6",
                "lineStyle": "solid",
                "lineWidth": "1px",
                "radius": Object {
                  "m": "6px",
                  "s": "4px",
                },
              },
            },
            "interaction": Object {
              "hover": Object {
                "animation": ".3s ease-out",
                "animationDuration": ".3s",
                "animationTiming": "ease-out",
                "feedbackOpacity": ".7",
              },
            },
            "palette": Object {
              "BACKGROUND": "#f5f6fa",
              "BORDER": "#d6d6d6",
              "BRAND": "#00c4cc",
              "COLUMN": "#f9f9f9",
              "DANGER": "#ef475b",
              "HEADER_GREEN": "#57d0d5",
              "MAIN": "#00a5ab",
              "OUTLINE": "rgba(0,165,171,0.5)",
              "OVERLAY": "rgba(0,0,0,0.15)",
              "SCRIM": "rgba(0,0,0,0.5)",
              "TEXT_BLACK": "#333",
              "TEXT_DISABLED": "#c1c1c1",
              "TEXT_GREY": "#767676",
              "TEXT_LINK": "#007bc2",
              "WARNING": "#ff8800",
              "disableColor": [Function],
              "hoverColor": [Function],
            },
            "size": Object {
              "font": Object {
                "GRANDE": 18,
                "SHORT": 11,
                "TALL": 14,
                "VENTI": 24,
              },
              "mediaQuery": Object {
                "SP": 599,
                "TABLET": 959,
              },
              "pxToRem": [Function],
              "space": Object {
                "L": 40,
                "M": 32,
                "S": 24,
                "XL": 48,
                "XS": 16,
                "XXL": 56,
                "XXS": 8,
              },
            },
          }
        }
      >
        <span
          className="sc-AxiKw kXHM active "
        />
      </styled.span>
      <styled.span
        themes={
          Object {
            "frame": Object {
              "border": Object {
                "default": "1px solid #d6d6d6",
                "lineStyle": "solid",
                "lineWidth": "1px",
                "radius": Object {
                  "m": "6px",
                  "s": "4px",
                },
              },
            },
            "interaction": Object {
              "hover": Object {
                "animation": ".3s ease-out",
                "animationDuration": ".3s",
                "animationTiming": "ease-out",
                "feedbackOpacity": ".7",
              },
            },
            "palette": Object {
              "BACKGROUND": "#f5f6fa",
              "BORDER": "#d6d6d6",
              "BRAND": "#00c4cc",
              "COLUMN": "#f9f9f9",
              "DANGER": "#ef475b",
              "HEADER_GREEN": "#57d0d5",
              "MAIN": "#00a5ab",
              "OUTLINE": "rgba(0,165,171,0.5)",
              "OVERLAY": "rgba(0,0,0,0.15)",
              "SCRIM": "rgba(0,0,0,0.5)",
              "TEXT_BLACK": "#333",
              "TEXT_DISABLED": "#c1c1c1",
              "TEXT_GREY": "#767676",
              "TEXT_LINK": "#007bc2",
              "WARNING": "#ff8800",
              "disableColor": [Function],
              "hoverColor": [Function],
            },
            "size": Object {
              "font": Object {
                "GRANDE": 18,
                "SHORT": 11,
                "TALL": 14,
                "VENTI": 24,
              },
              "mediaQuery": Object {
                "SP": 599,
                "TABLET": 959,
              },
              "pxToRem": [Function],
              "space": Object {
                "L": 40,
                "M": 32,
                "S": 24,
                "XL": 48,
                "XS": 16,
                "XXL": 56,
                "XXS": 8,
              },
            },
          }
        }
      >
        <span
          className="sc-AxhUy cfHhAU"
        >
          <Component
            color="#fff"
            name="fa-check"
            size={10}
          >
            <FaCheck
              aria-hidden={true}
              className=""
              color="#fff"
              focusable={false}
              role="img"
              size={10}
            >
              <IconBase
                aria-hidden={true}
                attr={
                  Object {
                    "viewBox": "0 0 512 512",
                  }
                }
                className=""
                color="#fff"
                focusable={false}
                role="img"
                size={10}
              >
                <svg
                  aria-hidden={true}
                  color="#fff"
                  fill="currentColor"
                  focusable={false}
                  height={10}
                  role="img"
                  size={10}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "#fff",
                    }
                  }
                  viewBox="0 0 512 512"
                  width={10}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaCheck>
          </Component>
        </span>
      </styled.span>
    </div>
  </styled.div>
</Component>
`;
